src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,273 @@
+
+    DetailedRootSection * eventSource; /**< A pointer to the detailed resolution of the cluster instance.  When NULL, the event
+                                          source is assumed to come from the resource equal to the local node ID, and from the
+                                          default instance of the cluster. */
+
+    PriorityLevel relatedPriority; /**< EventPriority of the Related Event Number.  When this event and the related event are of the
+                                           same priority, the field may be omitted.  A value of kPriorityLevel_Invalid implies the
+                                           absence of any related event. */
+
+    TimestampType timestampType; /**< An enum indicating if the timestamp is valid and its type. */
+
+    bool urgent; /**< A flag denoting that the event is time sensitive.  When set, it causes the event log to be flushed. */
+};
+
+/**
+ * @brief
+ *   Structure for copying event lists on output.
+ */
+
+struct EventLoadOutContext
+{
+    EventLoadOutContext(chip::TLV::TLVWriter & aWriter, PriorityLevel aPriority, uint32_t aStartingEventID);
+
+    chip::TLV::TLVWriter & mWriter;
+    PriorityLevel mPriority;
+    uint32_t mStartingEventNumber;
+    uint32_t mCurrentTime;
+    uint32_t mCurrentEventNumber;
+    uint64_t mCurrentUTCTime;
+    bool mFirstUtc;
+    bool mFirst;
+};
+
+/**
+ *  @brief
+ *    A function that supplies eventData element for the event logging subsystem.
+ *
+ *  Functions of this type are expected to provide the eventData
+ *  element for the event logging subsystem. The functions of this
+ *  type are called after the event subsystem has generated all
+ *  required event metadata. The function is called with a
+ *  chip::TLV::TLVWriter object into which it will emit a single TLV element
+ *  tagged kTag_EventData; the value of that element MUST be a
+ *  structure containing the event data. The event data itself must
+ *  be structured using context tags.
+ *
+ *  @sa PlainTextWriter
+ *  @sa EventWriterTLVCopy
+ *
+ *  @param[inout] aWriter A reference to the chip::TLV::TLVWriter object to be
+ *                         used for event data serialization.
+ *
+ *
+ *  @param[in]    apAppState  A pointer to an application specific context.
+ *
+ *  @retval #CHIP_NO_ERROR  On success.
+ *
+ *  @retval other           An appropriate error signaling to the
+ *                          caller that the serialization of event
+ *                          data could not be completed. Errors from
+ *                          calls to the aWriter should be propagated
+ *                          without remapping. If the function
+ *                          returns any type of error, the event
+ *                          generation is aborted, and the event is not
+ *                          written to the log.
+ *
+ */
+
+typedef CHIP_ERROR (*EventWriterFunct)(chip::TLV::TLVWriter & aWriter, void * apAppState);
+
+// internal API
```

Comment:
Call it `InternalLoggingBufferHandler` if internal?

Reply:
Done, thanks